docs/aws/audit/lambdamonitoring/rules/lambda_concurrency.mdx

```diff
@@ -22,6 +22,203 @@ CBP,RBI_MD_ITF
+To prevent the misconfiguration of not having concurrency configured for an AWS Lambda function using the AWS CLI, you can follow these steps:
+
+1. **Install and Configure AWS CLI:**
+   Ensure that the AWS CLI is installed and configured with the necessary permissions to manage Lambda functions.
+   ```sh
+   aws configure
+   ```
+
+2. **Set Reserved Concurrency:**
+   Use the `put-function-concurrency` command to set the reserved concurrency for your Lambda function. This limits the number of concurrent executions for the function.
+   ```sh
+   aws lambda put-function-concurrency --function-name <function-name> --reserved-concurrent-executions <number>
+   ```
+
+3. **Verify Concurrency Configuration:**
+   Confirm that the concurrency setting has been applied correctly by using the `get-function-concurrency` command.
+   ```sh
+   aws lambda get-function-concurrency --function-name <function-name>
+   ```
+
+4. **Monitor and Adjust as Needed:**
+   Regularly monitor the concurrency settings and adjust them as necessary to ensure optimal performance and cost management.
+   ```sh
+   aws lambda update-function-configuration --function-name <function-name> --reserved-concurrent-executions <new-number>
+   ```
+
+By following these steps, you can ensure that your AWS Lambda functions have the appropriate concurrency settings configured, preventing potential performance issues and resource exhaustion.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of not having concurrency configured in AWS Lambda using Python scripts, you can follow these steps:
+
+1. **Install AWS SDK for Python (Boto3):**
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip.
+
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials:**
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+   ```bash
+   aws configure
+   ```
+
+   Alternatively, you can set the environment variables directly in your script:
+
+   ```python
+   import os
+
+   os.environ['AWS_ACCESS_KEY_ID'] = 'your_access_key_id'
+   os.environ['AWS_SECRET_ACCESS_KEY'] = 'your_secret_access_key'
+   os.environ['AWS_DEFAULT_REGION'] = 'your_default_region'
+   ```
+
+3. **Create a Python Script to Configure Concurrency:**
+   Use Boto3 to configure the concurrency settings for your Lambda function. Below is a sample script to set the reserved concurrency for a Lambda function.
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon Lambda
+   client = boto3.client('lambda')
+
+   # Function to set concurrency
+   def set_lambda_concurrency(function_name, reserved_concurrent_executions):
+       response = client.put_function_concurrency(
+           FunctionName=function_name,
+           ReservedConcurrentExecutions=reserved_concurrent_executions
+       )
+       return response
+
+   # Example usage
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>